test/e2e/common/configmap.go

```diff
@@ -174,6 +181,19 @@ var _ = ginkgo.Describe("[sig-node] ConfigMap", func() {
 		}, metav1.CreateOptions{})
 		framework.ExpectNoError(err, "failed to create ConfigMap")
 
+		ginkgo.By("setting a watch for the ConfigMap")
+		// setup a watch for the ConfigMap
+		resourceWatch, err := f.ClientSet.CoreV1().ConfigMaps(testNamespaceName).Watch(context.TODO(), metav1.ListOptions{LabelSelector: "test-configmap-static=true", TimeoutSeconds: &generalWatchTimeout})
+		framework.ExpectNoError(err, "Failed to setup watch on newly created ConfigMap")
+
+		resourceWatchChan := resourceWatch.ResultChan()
+		ginkgo.By("waiting for the ConfigMap to be added")
+		for watchEvent := range resourceWatchChan {
+			if watchEvent.Type == watch.Added {
+				break
+			}
+		}
```

Comment:
At a minimum this should have a flag set to false before the loop, update the flag to true on the inside of the loop, and do a framework.Expect(...) after the loop, eg:
```
foundWatchDeleteEvent := false
for ... {
 if ... {
  foundWatchDeleteEvent := true
  break
  }
}
framework.ExpectEqual(true, foundWatchDeleteEvent, "expected to find a watch.Delete event configmap %s", configMap)
```

Reply:
Added in 7ba8b35b9c4